frontend/src/hooks/api/getters/useIncomingWebhookEvents/useIncomingWebhookEvents.ts

```diff
@@ -0,0 +1,79 @@
+import useSWRInfinite, {
+    SWRInfiniteConfiguration,
+    SWRInfiniteKeyLoader,
+} from 'swr/infinite';
+import { formatApiPath } from 'utils/formatPath';
+import handleErrorResponses from '../httpErrorResponseHandler';
+import useUiConfig from '../useUiConfig/useUiConfig';
+import { IIncomingWebhookEvent } from 'interfaces/incomingWebhook';
+import { useUiFlag } from 'hooks/useUiFlag';
+
+const ENDPOINT = 'api/admin/incoming-webhooks';
+
+type IncomingWebhookEventsResponse = {
+    incomingWebhookEvents: IIncomingWebhookEvent[];
+};
+
+const fetcher = async (url: string) => {
+    const response = await fetch(url);
+    await handleErrorResponses('Incoming webhook events')(response);
+    return response.json();
+};
+
+export const useIncomingWebhookEvents = (
+    incomingWebhookId?: number,
+    limit = 50,
+    options: SWRInfiniteConfiguration = {},
+) => {
+    const { isEnterprise } = useUiConfig();
+    const incomingWebhooksEnabled = useUiFlag('incomingWebhooks');
+
+    const getKey: SWRInfiniteKeyLoader = (
+        pageIndex: number,
+        previousPageData: IncomingWebhookEventsResponse,
+    ) => {
+        // Does not meet conditions
+        if (!incomingWebhookId || !isEnterprise || !incomingWebhooksEnabled)
```

Comment:
`!incomingWebhookId`
Is this a UI that can be opened without having an existing inc.webhook?

Reply:
I think this is related to: https://github.com/Unleash/unleash/pull/6317#discussion_r1500470878

The behavior we want for this hook is that you can call it whenever, but if you're not passing an `incomingWebhookId`, it should not fetch data.

Since you cannot conditionally call hooks in React, this matches the previous comment well, because you can just call this hook every time and it will gracefully handle undefined cases.